tests/eligibility/test_iae.py

```diff
@@ -535,6 +537,32 @@ def test_eligibility_diagnosis_certify_criteria(mocker, EligibilityDiagnosisFact
     assert updated_certification.certified_at > certification.certified_at
 
 
+@pytest.mark.parametrize(
+    "async_mode_only, certify_criteria_called, async_certify_criteria_called, side_effect",
+    [
+        pytest.param(True, False, True, None, id="async_mode_only"),
+        pytest.param(False, True, False, None, id="sync_mode_only"),
+        pytest.param(False, True, True, RetryTask, id="sync_mode_with_retry_task"),
+    ],
+)
+@freeze_time("2024-09-12")
+def test_eligibility_diagnosis_certify_criteria_retry_on_error(
+    settings, async_mode_only, certify_criteria_called, async_certify_criteria_called, side_effect
+):
+    eligibility_diagnosis = IAEEligibilityDiagnosisFactory(
+        certifiable=True, criteria_kinds=[AdministrativeCriteriaKind.RSA]
+    )
+
+    settings.CERTIFY_CRITERIA_ASYNC_MODE_ONLY = async_mode_only
+    with (
+        mock.patch("itou.eligibility.models.common.certify_criteria", side_effect=side_effect) as sync_mock,
+        mock.patch("itou.eligibility.models.common.async_certify_criteria") as async_mock,
+    ):
+        eligibility_diagnosis.certify_criteria()
+        assert sync_mock.called == certify_criteria_called
+        assert async_mock.called == async_certify_criteria_called
```

Comment:
_nit_, vu que c'est des booléens :
```suggestion
        assert sync_mock.called is certify_criteria_called
        assert async_mock.called is async_certify_criteria_called
```